### PR TITLE
Fixed runtime exception

### PR DESCRIPTION
**Pull request by tiomny**
No mapping to a relational type can be found for the CLR type 'SqlParameter[]

### CODE REVIEW

**Comment by tiomny** [comment]
Hold a minute - now I get "Value cannot be null. (Parameter 'parameters')"

**Comment by tiomny** [comment]
Sorry for the late full manual testing - it seems fine now. @ErikEJ could you merge it?

File: src/GUI/RevEng.Core/DbContextExtensions [hunk 1]
```diff
@@ -12,8 +12,10 @@ namespace #NAMESPACE#
 {
     public static class DbContextExtensions
     {
-        public static async Task<T[]> SqlQueryAsync<T>(this DbContext db, string sql, IEnumerable<object> parameters, CancellationToken cancellationToken = default) where T : class
+        public static async Task<T[]> SqlQueryAsync<T>(this DbContext db, string sql, object[] parameters = null, CancellationToken cancellationToken = default) where T : class
         {
+            parameters ??= new object[] { };
```
Comment (ErikEJ) [inline src/GUI/RevEng.Core/DbContextExtensions:17]: What is this syntax ? Will it work with C# 8 projects?

File: src/GUI/RevEng.Core/DbContextExtensions [hunk 2]
```diff
@@ -12,8 +12,10 @@ namespace #NAMESPACE#
 {
     public static class DbContextExtensions
     {
-        public static async Task<T[]> SqlQueryAsync<T>(this DbContext db, string sql, IEnumerable<object> parameters, CancellationToken cancellationToken = default) where T : class
```
Comment (ErikEJ) [inline src/GUI/RevEng.Core/DbContextExtensions:15]: Can we remove a using now?

**Review by ErikEJ — CHANGES_REQUESTED** [review]

File: src/GUI/RevEng.Core/DbContextExtensions (hunk 1 above)
Comment (tiomny) [reply src/GUI/RevEng.Core/DbContextExtensions:17]: > Available in C# 8.0 and later, the null-coalescing assignment operator ??= assigns the value of its right-hand operand to its left-hand operand only if the left-hand operand evaluates to null. The ??= operator doesn't evaluate its right-hand operand if the left-hand operand evaluates to non-null.

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator

**Comment by ErikEJ** [comment]
Assume you have run some integration tests now 😄

**Review by ErikEJ — APPROVED** [review]

**Comment by tiomny** [comment]
> 
> 
> Assume you have run some integration tests now 😄

This code is working in DEV environment :)

**Comment by ErikEJ** [comment]
Thanks, I have just updated the tool .zip files

**Pull request merged**